### PR TITLE
Modified DateField to allow input type to be set in props

### PR DESCRIPTION
**Pull request by jlrodney**
**Checklist**:

* [X] Documentation
* [X] Tests
* [X] Ready to be merged

### CODE REVIEW

**Comment by penx** [comment]
Deploy preview for *govuk-react* processing.

Building with commit ddb86fb7044a75c0cfd24fab3105d87296a14817

https://app.netlify.com/sites/govuk-react/deploys/5ce712b51fb9530008da5d80

**Comment by penx** [comment]
I'd ideally like to see an accompanying test.

Ideally with react-testing-library else with a snapshot.

Other than that, all good! Thanks for the contribution

**Comment by penx** [comment]
also please rebase of master to resolve the yarn.lock file

**Comment by jlrodney** [comment]
Ok great, I'll add a test. Should I squish my commits?

**Comment by jsrobertson** [comment]
Shouldn't be too much of a problem @jlrodney as they'll get squashed when merged.

File: components/date-field/src/input.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ class Input extends React.Component {
         <StyledInput
           name={name}
           error={error}
-          type="number"
+          type={this.props.type ? this.props.type : 'number'}
```
Comment (jsrobertson) [inline components/date-field/src/input.js]: `number` is already set in the default props.

**Comment by penx** [comment]
I think the intention here was so you don't need type=number, which I've now fixed in another PR so closing this

**Pull request closed without merging**